agents.py

```diff
@@ -510,14 +510,16 @@ def execute_action(self, agent, action):
             agent.direction += Direction.L
         elif action == 'Forward':
             agent.bump = self.move_to(agent, agent.direction.move_forward(agent.location))
-        #         elif action == 'Grab':
-        #             things = [thing for thing in self.list_things_at(agent.location)
-        #                     if agent.can_grab(thing)]
-        #             if things:
-        #                 agent.holding.append(things[0])
+        elif action == 'Grab':
+            things = [thing for thing in self.list_things_at(agent.location)]
+            if agent.can_grab(things[0]):
+                if things:
```

Comment:
This line `if things` should be on the top otherwise `things[0]` (accessed in the line above) would raise an exception!

Reply:
yes that is true